### ISSUE TITLE
Add ability to bind to NATS JetStream by stream name.

### ISSUE THREAD

**Issue by Jarema** [dapr/components-contrib#2161]
## Describe the feature
Currently, users of DAPR binding can subscribe to a stream only by looking up stream by matching subject.
I would like to add a config option to lookup stream by its name and then using it when calling `Subscribe`.
This is often simpler and more explicit and will additionaly allow fixing #2145 when user do not pass additional filter subject to the Consumer.

Let me know if you're ok with such approach, then I'll proceed with implementation.

## Release Note

RELEASE NOTE:

**Comment by NickLarsenNZ** [comment]
Could you give an example of what the code would would look like?

Eg, would the topic then be the stream name?

```py
@app.subscribe(pubsub_name='order-pub-sub', topic='orders')
def mytopic(event: v1.Event) -> None:
    data = json.loads(event.Data())
    logging.info('Subscriber received: ' + str(data))
```

Or do you mean the handlers would still discriminate on topic name, but the `Component` would force a particular stream?

**Comment by Jarema** [comment]
I was rather thinking of adding a new option to the `Component` config for NATS JetStream PubSub.

**Comment by yaron2** [comment]
@Jarema will you be working on this?

**Comment by NickLarsenNZ** [comment]
> I was rather thinking of adding a new option to the Component config for NATS JetStream PubSub.

So that was solved in #1114 right?
Is it still possible to then write different handlers to filter the subject (`topic` in Dapr) like in my example above?
